### DISCUSSION TITLE
Question to javascript library in footer

### DISCUSSION THREAD

**Discussion by WM-Loose** [templaza/astroid-framework#358] [General]
Back when Astroid was part of Joomdev, the developers thought they could load javascripts in the footer (not the header) to make the page load faster. But with that they created a big problem because they didn't use Joomla Core loading. For example, if an extension needed to use javascript, there was an error because Astroid loaded javascript in the footer (i.e. the script started before the library was loaded). The problem was solved by loading the library twice - once correctly with Core Joomla (which normally loads the library in the header) and once incorrectly in the footer of the template. Everything worked, but only at the cost of loading the same javascript library twice.

The question is, do you happen to know if this problem has already been solved?

**Comment by sonvnn** [comment]
Hello,

You can go to Administrator, switch load jQuery and Bootstrap from Astroid to Joomla. The system will load core JS in the head.  
![Screen Shot 2023-07-24 at 10 30 01](https://github.com/templaza/astroid-framework/assets/3204103/879f7fb7-9e90-4f8f-b506-9e9b42b25b3e)

Best Regards,
Sonny

**Comment by WM-Loose** [comment]
Hi Sonny, thank you very much.
